wave_kinematics/include/wave/kinematics/rotation.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef __WAVE_KINEMATICS_ROTATION_HPP__
+#define __WAVE_KINEMATICS_ROTATION_HPP__
+
+#include <iostream>
+#include <functional>
+
+#include "wave/utils/utils.hpp"
+#include <kindr/Core>
+
+namespace wave {
+
+class Rotation {
+public:
+  // Internal storage as a kindr Rotation object.
+  kindr::RotationMatrixD rotation_object;
+  // The comparision threshold used to test if rotation objects are
+  // "close enough" to each other.  Used in unit testing and the "isNear"
+  // function.
+  static double comparision_threshold;
+
+public:
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+
+  // Rotation Object Initializers.
+  Rotation();
+  static bool fromEulerXYZ(const double rotation_x,
```

Comment:
It does touch the object. You can tell because there is no `const` on the end.

Returning a reference is a common pattern that allows chaining, and is used in Eigen. For example 
`r.setFromEulerXyz(x).invert()`.